### PR TITLE
ingester: promote owned series stable and enable by default

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

Preparing for the 2.14 release. We've been running this in production in GL and have since fixed the rough edges of this feature.

#### Which issue(s) this PR fixes or relates to

related to https://github.com/grafana/mimir/issues/7847

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * [CHANGE] Store-gateway: enabled `-blocks-storage.bucket-store.max-concurrent-queue-timeout` by default with a timeout of 5 seconds. #8496
 * [CHANGE] Store-gateway: enabled `-blocks-storage.bucket-store.index-header.lazy-loading-concurrency-queue-timeout` by default with a timeout of 5 seconds . #8667
 * [CHANGE] Distributor: Incoming OTLP requests were previously size-limited by using limit from `-distributor.max-recv-msg-size` option. We have added option `-distributor.max-otlp-request-size` for limiting OTLP requests, with default value of 100 MiB. #8574
+* [CHANGE] Ingester: enable owned series to stable and enable by default. Active series are now updated along with owned series. They decrease when series change ownership between ingesters. This helps provide a more accurate total of active series when ingesters are added. #8695
```
Comment (gotjosh) [inline CHANGELOG.md:13]:
```suggestion
* [CHANGE] Ingester: Promote `-ingester.track-ingester-owned-series` to stable and enabled by default. Active series are now updated along with owned series. They decrease when series change ownership between ingesters. This helps provide a more accurate total of active series when new ingesters are added. #8695
```

**Review by gotjosh — APPROVED** [review]
LGTM

**Comment by dimitarvdimitrov** [comment]
feedback from @pr00se 

> i don't think we can do this until we work around [this limitation](https://github.com/grafana/dskit/blob/main/ring/token_range.go#L72-L73)

https://github.com/grafana/mimir/blob/42ccb78851f02cfe70e5681332fc2486b206a1f3/vendor/github.com/grafana/dskit/ring/token_range.go#L72-L77

**Comment by dimitarvdimitrov** [comment]
from a slack conversation with @pr00se 

> can we enable it by default only when zone awareness is enabled - anything tricky there?


> the only other caveat is the replication factor == number of zones









> i'm not actually sure what the failure mode is if it is misconfigured. i think it might just throw an error in the logs and then not recalculate owned series, so if you're limiting based on them you just have no limits?
> and it should probably have some documentation somewhere too


because of these we agreed to keep the feature as experimental.

**Pull request closed without merging**